### ISSUE TITLE
[Bug]: Cant fix the random seed.

### ISSUE THREAD

**Issue by koonimaru** [experimental-design/bofire#629]
### What happened?

I tried the following code from your tutorial. I added random seeds but that did not fix the result.

### Please provide a minimal, reproducible example of the unexpected behavior.

`
import numpy as np
import random
import torch
torch.manual_seed(0)
random.seed(0)
np.random.seed(0)
torch.backends.cudnn.deterministic=True
torch.cuda.manual_seed(0)
torch.backends.cudnn.benchmark = False

domain = Domain(
    inputs=[ContinuousInput(key=f"x{i+1}", bounds=(0, 1)) for i in range(2)],
    outputs=[ContinuousOutput(key="y")],
)
model_type = "fully-quadratic"
n_experiments = 13

fig, ax = plt.subplots(figsize=(4, 4))

for crit, label in [
    (DOptimalityCriterion, "D-Optimality"),
    (AOptimalityCriterion, "A-Optimality"),
    (KOptimalityCriterion, "K-Optimality"),
    (EOptimalityCriterion, "E-Optimality"),
    (IOptimalityCriterion, "I-Optimality"),
]:
    criterion = crit(formula=model_type)
    data_model = DoEStrategy(
        domain=domain,
        criterion=criterion,
        ipopt_options={"max_iter": 300},
        seed=0,
    )
    strategy = strategies.map(data_model=data_model)
    design = strategy.ask(candidate_count=n_experiments)
    obj_value = get_objective_function(
        criterion=criterion, domain=domain, n_experiments=n_experiments
    ).evaluate(design.to_numpy().flatten())
    ax.scatter(design.x1, design.x2, s=40, label=f"{label}")

ax.set_title("Designs with different optimality criteria")
ax.set_xlabel("$x_1$")
ax.set_ylabel("$x_2$")
ax.grid(alpha=0.3)
ax.legend()

plt.show()
`

### Please paste any relevant traceback/logs produced by the example provided.

```shell

```

### BoFire Version

0.2.2

### Python Version

3.11

### Operating System

ubuntu

**Comment by MyonicS** [comment]
@Osburg I think @koonimaru is trying to get a deterministic result from the sample code. i.e. it would be nice to have a way to set up code in a way so that I get the same design each time it is run when specifically handing it a seed. I had a similar issue but i did not tinker with it too much.

**Comment by Osburg** [comment]
Hey @MyonicS,

The seed of the DoEStrategy was not used during sampling the initial guess for the DoE problem. The PR above should fix that. Thanks! :)

Cheers
Aaron
